src/transpilation/resolve.ts

```diff
@@ -0,0 +1,206 @@
+import * as path from "path";
+import * as resolve from "enhanced-resolve";
+import * as ts from "typescript";
+import * as fs from "fs";
+import { EmitHost, ProcessedFile } from "./utils";
+import { SourceNode } from "source-map";
+import { getEmitPathRelativeToOutDir, getProjectRoot, getSourceDir } from "./transpiler";
+import { formatPathToLuaPath } from "../utils";
+import { couldNotReadDependency, couldNotResolveRequire } from "./diagnostics";
+
+const resolver = resolve.ResolverFactory.createResolver({
+    extensions: [".lua"],
+    enforceExtension: true, // Resolved file must be a lua file
+    fileSystem: { ...new resolve.CachedInputFileSystem(fs) },
+    useSyncFileSystemCalls: true,
+});
+
+interface ResolutionResult {
+    resolvedFiles: ProcessedFile[];
+    diagnostics: ts.Diagnostic[];
+}
+
+// Cache for getting source files from the program
+const projectFileCache = new Set<string>();
```

Comment:
Is there any good way to avoid having this in global/module scope? maybe store it in the transformation context?

Reply:
Removed this completely